phosphorus/parse.py

```diff
@@ -112,6 +112,17 @@ def mylog(s): log(s,"Span.update")
                     s = str(item) #Problem if we want to use keywords as ConstantVals?
                     mylog(f"|{item}| transforms to |{s}|")
                     item = Span.parse(spaces + s)
+                    # logic for add parens if necessary
+                    outerParens = ( # check for surrounding delimiters in self
+                            self[n - 1].string in Token.delims and
+                            len(self) > n + 1 and
+                            self[n + 1].string == Token.delims[self[n - 1].string]
+                    )
+                    onlyItem = len(self) == 2 and n == 0 # check if result of substitution is the only thing in self
+                    # if len == 1, then there is only one token in item, so parens aren't needed
+                    if not (outerParens or onlyItem or len(item) == 1):
+                        # add surrounding parens if needed
+                        item = Span.parse(spaces + "(" + s + ")")
                     mylog(f"Parsed: |{item}|")
                     if item.printlen() == 1: item = item[0]
```

Comment:
I believe that this line removes parentheses from single-item Spans -- maybe worth testing how it interacts with the logic above?